model/task_lifecycle.go

```diff
@@ -778,7 +778,29 @@ func removeNextMergeTaskDependency(cq commitqueue.CommitQueue, currentIssue stri
 	if err != nil {
 		return errors.Wrap(err, "unable to find current merge task")
 	}
-	return errors.Wrap(nextMerge.RemoveDependency(currentMerge.Id), "unable to remove dependency")
+	if err = nextMerge.RemoveDependency(currentMerge.Id); err != nil {
+		return errors.Wrap(err, "unable to remove dependency")
+	}
+
+	if currentIndex > 0 {
```

Comment:
If something gets removed from the commit queue, we should both remove the dependency of the merge task behind it on what was removed (already here), and make the merge task behind it now depend on the merge task in front of it (added here). Or to put it differently, we're doing what happens when you remove something in the middle of a linked list

Reply:
This makes sense, but can you add a comment or change the function name? It's not intuitive from the title "removeNextMergeTaskDependency"

Reply:
added a comment